cpp/src/quantiles/quantiles.hpp

```diff
@@ -80,5 +87,105 @@ void midpoint(int64_t& result, int64_t lhs, int64_t rhs)
 }
 
 } // end of namespace interpolate
+
+namespace detail {
+
+/**
+ * @brief Helper struct that calculates the values needed to get quantile values
+ * by interpolation.
+ * 
+ * For a quantile that lies between indices i and j, this struct calculates 
+ * i (lower_bound),
+ * j (upper_bound),
+ * index nearest to quantile between i and j (nearest),
+ * and the fractional distance that the quantile lies ahead i (fraction)
+ * 
+ */
+struct quantile_index {
```

Comment:
Why is this a struct and not just a function? Just return the four values it computes. They are only used locally.

Reply:
Return it how? It'd be better to return them as a struct rather than a tuple because it is relatively cumbersome to access tuple elements. With structs you can simply access its members. And it I were to make a function that returns said struct, why shouldn't it just be a member of the struct itself. And then, it makes the most sense to make it a constructor of that struct. I think this is what went through @kovaltan's mind when he made this. And these are the thoughts that went through my mind when I thought of refactoring it so I just let it be.